### PR TITLE
DAOS-2289 ctrl: Add PCI whitelist option for SPDK setup with prep-nvme

### PR DESCRIPTION
**Pull request by sydneyvanda**
Added option to include PCI whitelist to specify devices to be
unbound by SPDK, with "-w PCI_ADDR" option (--pci-whitelist for long option)
Avoids all of the uio devices being unbound by default as well
"$sudo daos_server storage prep-nvme -w 0000:81:00.0 -p 4096 -u smvanda"

Updates to README - fixed error with setup_spdk.sh location and
command for running prep-nvme script.

Signed-off-by: Sydney Vanda <sydney.m.vanda@intel.com>

### CODE REVIEW

File: src/control/server/README.md [hunk 1]
```diff
@@ -109,9 +109,9 @@ TODO: examples for both DCPM and RAM (emulation) SCM classes including config fi
...
 
-The configuration commands that require elevated permissions are in `src/control/mgmt/init/setup_spdk.sh` (script is installed as `install/share/setup_spdk.sh`).
+The configuration commands that require elevated permissions are in `src/control/server/init/setup_spdk.sh` (script is installed as `install/share/control/setup_spdk.sh`).
 
 The sudoers file can be accessed with command `visudo` and permissions can be granted to a user to execute a specific command pattern (requires prior knowledge of `daos_server` binary location):
 ```
```
Comment (sydneyvanda) [inline src/control/server/README.md]: update line 121: daos_server storage prep-nvme --help

File: src/control/server/storage_nvme.go [hunk 2]
```diff
@@ -112,6 +115,39 @@ func (s *spdkSetup) prep(nrHugepages int, usr string) error {
 		hPages, tUsr, stderr.String())
 }
 
+// prep executes setup script to allocate hugepages and unbind specified PCI
+// devices (that don't have active mountpoints) from generic kernel driver to be
+// used with SPDK.
+//
+// NOTE: will make the controller disappear from /dev until reset() called.
+func (s *spdkSetup) prep(nrHugepages int, usr string, wlist string) error {
```
Comment (tanabarr) [inline src/control/server/storage_nvme.go]: probably want to consolidate the two commands, prepAll could call prep with empty wlist string and prep only populate wlist env if wlist parameter is not empty? does that make sense?

Comment (sydneyvanda) [reply src/control/server/storage_nvme.go]: yes definitely I can simplify the code, just wanted to make sure that this seemed like a good addition to prep-nvme command?

Comment (tanabarr) [reply src/control/server/storage_nvme.go]: yes looks good, might be worth adding a description of what the whitelist syntax looks like when it's not just a single PCI 
 address

File: src/control/server/README.md [hunk 3]
```diff
@@ -109,16 +109,16 @@ TODO: examples for both DCPM and RAM (emulation) SCM classes including config fi
...
+NVMe access through SPDK as an unprivileged user can be enabled by first running `sudo daos_server storage prep-nvme -w 0000:81:00.0 -p 4096 -u bob`. This will perform the required setup in order for `daos_server` to be run by user "bob" who will own the hugepage mountpoint directory and vfio groups as needed in SPDK operations. If the `target-user` is unspecified (`-u` short option), the target user will be the issuer of the sudo command (or root if not using sudo). The specification of `hugepages` (`-p` short option) defines the number of huge pages to allocate for use by SPDK. The specification of `pci-whitelist` (`-w` short option) allows user to optionally specify which PCI devices to unbind from the Kernel driver for use with SPDK, as opposed to unbinding all devices by default.
```
Comment (kjacque) [inline src/control/server/README.md]: Might be nice to have an example of multiple items in the whitelist.

File: src/control/server/commands.go [hunk 4]
```diff
@@ -102,7 +102,8 @@ func (s *ListStorCmd) Execute(args []string) (errs error) {
 // PrepNvmeCmd is the struct representing the command to prep NVMe SSDs
 // for use with the SPDK as an unprivileged user.
 type PrepNvmeCmd struct {
-	NrHugepages int    `short:"p" long:"hugepages" description:"Number of hugepages to allocate for use by SPDK (default 1024)"`
+	PCIWhiteList string `short:"w" long:"pci-whitelist" description:"Specify PCI devices to be unbound from Kernel driver and used with SPDK (default is all PCI devices). Must be specified as a whitespace separated list of full PCI addresses (-w \"0000:81:00.0 000:2\"). If one of the addresses is non-valid (000:2), then that device will be skipped, unless it is the only address listed in which case all PCI devices will be blacklisted."`
```
Comment (kjacque) [inline src/control/server/commands.go]: This line is so wide it's kind of unwieldy. Possible to break up this string literal into segments for code readability? In Go I believe you could do something like:

```
"Here is my string literal. " +
"This is the other half of it."
```

Comment (liw) [reply src/control/server/commands.go]: If these descriptions will be printed for "help", then we should keep them short anyway, shouldn't we...?

Comment (sydneyvanda) [reply src/control/server/commands.go]: I think you are right, I simplified the output significantly and added more detail in the README instead.

Comment (liw) [reply src/control/server/commands.go]: Thanks.

**Pull request merged**